prov/util/src/util_coll.c

```diff
@@ -1178,31 +1178,37 @@ int ofi_av_set(struct fid_av *av, struct fi_av_set_attr *attr,
 {
 	struct util_av *util_av = container_of(av, struct util_av, av_fid);
 	struct util_av_set *av_set;
-	int ret, iter;
+	uint64_t offset;
+	int ret;
 
 	if (!util_av->coll_mc) {
 		ret = util_coll_av_init(util_av);
 		if (ret)
 			return ret;
 	}
 
-	av_set = calloc(1,sizeof(*av_set));
+	av_set = calloc(1, sizeof(*av_set));
 	if (!av_set)
 		return -FI_ENOMEM;
 
 	ret = util_av_set_init(av_set, util_av, context);
 	if (ret)
 		goto err1;
 
-	av_set->fi_addr_array = calloc(ofi_av_size(util_av),
+	size_t max_size = attr->count;
+	if (max_size == 0)
+		max_size = ofi_av_size(util_av);
+	av_set->fi_addr_array = calloc(max_size,
 				       sizeof(*av_set->fi_addr_array));
 	if (!av_set->fi_addr_array)
 		goto err2;
 
-	for (iter = 0; iter < attr->count; iter++) {
-		av_set->fi_addr_array[iter] =
-			util_av->coll_mc->av_set->fi_addr_array[iter * attr->stride];
-		av_set->fi_addr_count++;
+	for (offset = attr->start_addr;
+	     offset <= attr->end_addr;
+	     offset += attr->stride) {
+		assert(av_set->fi_addr_count < max_size);
+		av_set->fi_addr_array[av_set->fi_addr_count++] =
+			util_av->coll_mc->av_set->fi_addr_array[offset];
```

Comment:
The changes to libfabric should be separated from a commit that updates fabtests.

Reply:
Yes, please separate.  We'll want to pull in the fix to libfabric into v1.13.x branch, but not the new testing.

Reply:
Done the associated fabtests is now in PR #7141